hastexo/common.py

```diff
@@ -238,6 +246,15 @@ def get_stack(name, course_id, student_id, prop=None):
         return stack
 
 
+# If ServerSelectionTimeoutError is raised here, try again (max attempts = 3)
+# Before every subsequent retry, wait
+# Use before_sleep_log to log attempts
+# Reraise the exception from last attempt
+@retry(retry=retry_if_exception_type(ServerSelectionTimeoutError),
+       stop=stop_after_attempt(3),
+       wait=wait_exponential(),
+       before_sleep=before_sleep_log(logger, logging.WARNING),
+       reraise=True)
```

Comment:
Do I understand correctly that the default initial base with `wait_exponential()` is 2, so this means that we'll retry after 1 (2⁰) seconds, then 2 (2¹), then 4 (2²), and if it still doesn't succeed we give up?

Reply:
Also, do we only want to trigger the retry on `ServerSelectionTimeoutError`? Or do we perhaps want to expand this to all `PyMongoError` exceptions? (It's a read operation; redoing it in case of an error should kind of not hurt for any reason, unless I am missing something.)